src/main/java/uk/nhs/digital/nhsconnect/lab/results/translator/mapper/PatientMapper.java

```diff
@@ -1,12 +1,75 @@
 package uk.nhs.digital.nhsconnect.lab.results.translator.mapper;
 
+import org.hl7.fhir.dstu3.model.DateType;
+import org.hl7.fhir.dstu3.model.Enumerations;
+import org.hl7.fhir.dstu3.model.HumanName;
+import org.hl7.fhir.dstu3.model.Identifier;
 import org.hl7.fhir.dstu3.model.Patient;
 import org.springframework.stereotype.Component;
 import uk.nhs.digital.nhsconnect.lab.results.model.edifact.Message;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.PersonName;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.segmentgroup.InvestigationSubject;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.segmentgroup.PatientDetails;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.segmentgroup.ServiceReportDetails;
+
+import java.util.Optional;
 
 @Component
 public class PatientMapper {
+
+    protected static final String NHS_NUMBER_SYSTEM = "https://fhir.nhs.uk/Id/nhs-number";
+
     public Patient mapToPatient(final Message message) {
-        return new Patient();
+
+        final PatientDetails patientDetails = Optional.ofNullable(message.getServiceReportDetails())
+            .map(ServiceReportDetails::getSubject)
+            .map(InvestigationSubject::getDetails)
+            .orElseThrow(() -> new FhirValidationException("Unable to map message to patient details"));
+
+        final Patient patient = new Patient();
+
+        final PersonName personName = patientDetails.getName();
+        Optional.ofNullable(personName).ifPresent(name -> {
+            mapIdentifier(name, patient);
+            mapName(name, patient);
+        });
+
+        mapGender(patientDetails, patient);
+        mapDateOfBirth(patientDetails, patient);
+
+        return patient;
+    }
+
+    private void mapIdentifier(final PersonName name, final Patient patient) {
+        final Identifier identifier = new Identifier();
+
+        identifier.setSystem(NHS_NUMBER_SYSTEM);
+        Optional.ofNullable(name.getNhsNumber()).ifPresent(identifier::setValue);
```

Comment:
If `getNhsNumber()` returns `null` is there any harm in calling `identifier.setValue(null)`? Just wondering if there's any need to cycle through an `Optional` here

Reply:
Probably not 👍